Library/Source/BGFXCallback.cpp

```diff
@@ -0,0 +1,103 @@
+#include "BGFXCallback.h"
+#include <bx/bx.h>
+#include <bx/string.h>
+#include <bx/platform.h>
+#include <bx/debug.h>
+#include <stdarg.h>
+#include <bgfx/bgfx.h>
+
+namespace Babylon
+{
+    std::vector<uint8_t> BGFXCallback::screenShotBitmap;
+
+    void BGFXCallback::trace(const char* _filePath, uint16_t _line, const char* _format, ...)
+    {
+        va_list argList;
+        va_start(argList, _format);
+
+        traceVargs(_filePath, _line, _format, argList);
+
+        va_end(argList);
+    }
+
+    void BGFXCallback::fatal(const char* filePath, uint16_t line, bgfx::Fatal::Enum code, const char* str)
+    {
+        if (bgfx::Fatal::DebugCheck == code)
+        {
+            bx::debugBreak();
+        }
+        else
+        {
+            trace(filePath, line, "BGFX 0x%08x: %s\n", code, str);
+            BX_UNUSED(code, str);
+            abort();
+        }
+    }
+
+    void BGFXCallback::traceVargs(const char* filePath, uint16_t line, const char* format, va_list argList)
+    {
+        char temp[2048];
```

Comment:
nit: Magic number? Split out into `constexpr` variable?

Reply:
I don't see the point to use a constexpr here just for that value.

Reply:
Just so that that number can be put in a place that's easier to see from the top, and can be named. It's a nit, though, personal preference, not a problem either way.